torch_geometric/sampler/utils.py

```diff
@@ -27,16 +28,32 @@ def to_csc(
     perm: Optional[Tensor] = None
 
     if hasattr(data, 'adj'):
+        if src_node_time is not None:
+            raise NotImplementedError("Temporal sampling via 'SparseTensor' "
+                                      "format not yet supported")
         colptr, row, _ = data.adj.csc()
 
     elif hasattr(data, 'adj_t'):
+        if src_node_time is not None:
+            raise NotImplementedError("Temporal sampling via 'SparseTensor' "
+                                      "format not yet supported")
         colptr, row, _ = data.adj_t.csr()
 
     elif data.edge_index is not None:
         (row, col) = data.edge_index
-        if not is_sorted:
+        if not is_sorted and src_node_time is None:
             perm = (col * data.size(0)).add_(row).argsort()
             row = row[perm]
+        if not is_sorted and src_node_time is not None:
+            # Multiplying by raw `datetime[64ns]` values may cause overflows.
+            # As such, we normalize time into range [0, 1) before sorting:
+            src_node_time = src_node_time.to(torch.double, copy=True)
+            min_time, max_time = src_node_time.min(), src_node_time.max() + 1
+            src_node_time.sub_(min_time).div_(max_time)
+
+            perm = src_node_time[row].add_(col.to(torch.double)).argsort()
```

Comment:
```suggestion
            perm = col.to(torch.double).add_(src_node_time[row]).argsort()
```
Nitpick: makes it look similar to the computation when `src_node_time is None`.